core/dataURI/dataURI.go

```diff
@@ -144,12 +164,18 @@ func StoreDataFromTempData(uri string) common.SyncServiceError {
 	if trace.IsLogging(logger.TRACE) {
 		trace.Trace("Storing data from temp data at %s", uri)
 	}
+
 	dataURI, err := url.Parse(uri)
 	if err != nil || !strings.EqualFold(dataURI.Scheme, "file") {
 		return &Error{"Invalid data URI"}
 	}
 
-	tmpFilePath := dataURI.Path + ".tmp"
+	baseFilePath, err := filepath.Abs(filepath.Clean(dataURI.Path))
+	if err != nil {
+		return &Error{"Failed to resolve file path"}
+	}
+
+	tmpFilePath := baseFilePath + ".tmp"
 
 	if err := os.Rename(tmpFilePath, dataURI.Path); err != nil {
```

Comment:
`os.Rename(tmpFilePath, dataURI.Path)` -> `os.Rename(tmpFilePath, baseFilePath)`

Reply:
Please go through all the places which calls "rename" function, to make sure it rename to the correct file path